File: src/main/java/net/slipp/service/tag/TagHelper.java
package net.slipp.service.tag;

import java.util.Set;

import com.google.common.base.Function;
import com.google.common.base.Joiner;
import com.google.common.collect.Collections2;

import net.slipp.domain.tag.Tag;

public class TagHelper {

	public static String denormalizedTags(Set<Tag> tags) {
		if (tags == null) {
            return null;
        }

        Function<Tag, String> tagToString = new Function<Tag, String>() {
            @Override
            public String apply(Tag input) {
                return input.getName();
            }
        };

        return Joiner.on(",").join(Collections2.transform(tags, tagToString));
	}
}

File: src/main/scala/net/slipp/domain/qna/TagHelper.scala
package net.slipp.domain.qna

import java.util.Set
import com.google.common.base.Function
import com.google.common.base.Joiner
import com.google.common.collect.Collections2
import net.slipp.domain.tag.Tag

object TagHelper {
  def denormalizedTags(tags: Set[Tag]): String = {
    if (tags == null) {
      return null
    }
    val tagToString: Function[Tag, String] = new Function[Tag, String]() {
      def apply(input: Tag): String = {
        return input.getName
      }
    }
    return Joiner.on(",").join(Collections2.transform(tags, tagToString))
  }
}

